rpyc/rpyc_client/command_handlers.py

```diff
@@ -106,23 +118,24 @@ def __parse_input(self, command_with_arguments):
         return args.path
 
     def execute(self, command_with_arguments):
-        if len(command_with_arguments) < 2:
-            raise TypeError('expects at least one argument')
-
         if len(command_with_arguments) == 3:
+            # the command is: kill -signo pid
+            if not command_with_arguments[1].startswith('-'):
+                # signo should start with '-' prefix
+                raise ValueError('Incorrect usage')
             return self.__execute_kill_signo_pid(command_with_arguments)
 
-        if len(command_with_arguments) != 2:
-            raise TypeError("Incorrect usage")
-
-        if command_with_arguments[1] == 'all':
-            return self.__execute_kill_all()
+        if len(command_with_arguments) == 2:
+            if command_with_arguments[1] == 'all':
+                # the command is: kill all
+                return self.__execute_kill_all()
+            else:
+                # the command is kill pid
+                return self.__execute_kill_pid(command_with_arguments)
 
-        return self.__execute_kill_pid(command_with_arguments)
+        raise TypeError('Incorrect usage')
 
     def __execute_kill_signo_pid(self, command_with_arguments):
```

Comment:
would make more sense if this function got the signo directly, and wouldn't have to parse the args
that way this function can be used in other scenerios too

Reply:
That's the kill function. This function only parses and call the kill function